sycl/source/detail/device_impl.cpp

```diff
@@ -546,6 +546,23 @@ bool device_impl::has(aspect Aspect) const {
             sizeof(pi_bool), &support, nullptr) == PI_SUCCESS;
     return call_successful && support;
   }
+  case aspect::ext_oneapi_cubemap: {
+    pi_bool support = PI_FALSE;
+    bool call_successful =
+        getPlugin()->call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
```

Comment:
Is this `nocheck` on purpose? Why does it need to be `nocheck`?

Reply:
This is what is used elsewhere in this file to query for device aspects so I'm just following the norm. 

However, if I was to guess, it would be that failing a query for a device aspect is not a critical failure and so reporting a `false` is appropriate.

Reply:
I believe there is a good reason for `call` rather than `call_nocheck` in this case. Historically, it was useful as a lot of the device info enumerators were 1:1 with OpenCL and were passed directly to the OpenCL implementation. If the implementations did not recognize the enum value, the backend would return an error code. Though that is unlikely to be how these enums are implemented in UR now, having error codes indicate that the backend doesn't support the feature is a conservative solution and could also let newer DPC++ implementations use older UR plugins (I don't think we consider that valid in general though.)

Reply:
I've switched to using `call` rather than `call_nocheck`.

Reply:
Sorry, I switched the two around, I meant that `call_nocheck` had a benefit over `call` here. I would use coffee as an excuse, but 1PM feels a little too late for that. Once again, I apologize for the confusion.

Reply:
Reviving this thread as I've found that using `call` breaks all E2E tests on platforms that don't support all of our aspects.

My [sampled fetch PR](https://github.com/intel/llvm/pull/12447) was failing CI until I switched back to using `call_nocheck`.

For the moment I'd prefer to revert this and use `call_nocheck` like all the other queries in the file. I think changing all the queries to use `call` and work properly should be done in a separate PR, as the mechanism for catching exceptions from the `call` needs to be looked at as well.

What are your thoughts @steffenlarsen ?

Reply:
@przemektmalon - I noticed your comment on your other PR and noticed I had switched the two around when I made my previous comment. I'm sorry for the confusion, but I am with you on using `call_nocheck`.